addons/mail/models/discuss/discuss_channel_rtc_session.py

```diff
@@ -38,22 +45,46 @@ def create(self, vals_list):
         return rtc_sessions
 
     def unlink(self):
-        channels = self.channel_id
-        for channel in channels:
-            if channel.rtc_session_ids and len(channel.rtc_session_ids - self) == 0:
-                # If there is no member left in the RTC call, all invitations are cancelled.
-                # Note: invitation depends on field `rtc_inviting_session_id` so the cancel must be
-                # done before the delete to be able to know who was invited.
-                channel._rtc_cancel_invitations()
         notifications = [(channel, 'discuss.channel/rtc_sessions_update', {
             'id': channel.id,
             'rtcSessions': [('DELETE', [{'id': session_data['id']} for session_data in sessions_data])],
         }) for channel, sessions_data in self._mail_rtc_session_format_by_channel().items()]
+        session_ids_by_channel_by_url = defaultdict(lambda: defaultdict(list))
         for rtc_session in self:
             target = rtc_session.guest_id or rtc_session.partner_id
-            notifications.append((target, 'discuss.channel.rtc.session/ended', {'sessionId': rtc_session.id}))
-        self.env['bus.bus']._sendmany(notifications)
-        return super().unlink()
+            notifications.append((target, "discuss.channel.rtc.session/ended", {"sessionId": rtc_session.id}))
+            sfu_channel_uuid = rtc_session.channel_id.sfu_channel_uuid
+            url = rtc_session.channel_id.sfu_server_url
+            if not sfu_channel_uuid or not url:
+                continue
+            session_ids_by_channel_by_url[url][sfu_channel_uuid].append(rtc_session.id)
+        channels = self.channel_id
+        for channel in channels:
+            if channel.rtc_session_ids and len(channel.rtc_session_ids - self) == 0:
+                # If there is no member left in the RTC call, all invitations are cancelled.
+                # Note: invitation depends on field `rtc_inviting_session_id` so the cancel must be
+                # done before the unlink to be able to know who was invited.
+                channel._rtc_cancel_invitations()
+                # If there is no member left in the RTC call, we remove the SFU channel uuid as the SFU
+                # server will timeout the channel. It is better to obtain a new channel from the SFU server
+                # than to attempt recycling a possibly stale channel uuid.
+                channel.sfu_channel_uuid = False
+                channel.sfu_server_url = False
+        return_value = super().unlink()
+        self.env["bus.bus"]._sendmany(notifications)
+        if self.env.context.get('notify_rtc_server'):
```

Comment:
Just making sure, as context can be given from client, is it safe to allow it?
Why not always do it actually?

Reply:
for the context it is safe as it will only pass the ids of sessions that have been unlinked

It is not done everytime because in the other cases, the clients are either willingly closing the connection with the server, or the server will timeout these sessions (browser crash).